staging/src/k8s.io/kubectl/pkg/cmd/cmd.go

```diff
@@ -362,28 +362,36 @@ func NewKubectlCommand(o KubectlOptions) *cobra.Command {
 
 	flags.BoolVar(&warningsAsErrors, "warnings-as-errors", warningsAsErrors, "Treat warnings received from the server as errors and exit with a non-zero exit code")
 
-	pref := kuberc.NewPreferences()
-	if !cmdutil.KubeRC.IsDisabled() {
-		pref.AddFlags(flags)
-	}
-
 	kubeConfigFlags := o.ConfigFlags
 	if kubeConfigFlags == nil {
 		kubeConfigFlags = defaultConfigFlags().WithWarningPrinter(o.IOStreams)
 	}
 	kubeConfigFlags.AddFlags(flags)
+
+	pref := kuberc.NewPreferences()
```

Comment:
What is the reason of changing the order and moving from line:365-368 to here?

Reply:
reverted